### PR TITLE
Step 1 Initial Setup

### PR DESCRIPTION
**Pull request by a-reznikov**
image

- [x] Изучить документацию NextJS + go through[ https://nextjs.org/learn/foundations/about-nextjs](https://nextjs.org/learn/foundations/about-nextjs)
- [x] Create public GitHub repository with name like nextjs-learning
- [x] Create NextJS (/pages) project with typescript and tailwind CSS
- [x] Create simple header footer components
- [x] Create commit and PR to main branch
- [x] Send link to PR to your lead in slack

### CODE REVIEW

File: src/components/common/footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import Image from "next/image";
+
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: remove empty lines

File: src/styles/fonts.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const OPEN_SANS_400 = Open_Sans({
+  subsets: ["latin"],
+  weight: "400",
+});
+
+const OPEN_SANS_600 = Open_Sans({
+  subsets: ["latin"],
+  weight: "600",
+});
```
Comment (Ilya-Valasiuk) [inline src/styles/fonts.ts]: https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

Comment (Ilya-Valasiuk) [reply src/styles/fonts.ts]: src/fonts/index.tsx

File: src/components/common/footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { FOOTER_NAVIGATION, SOCIAL_LINKS } from "@/constants/links";
+
+import { OPEN_SANS_400 } from "@/styles/fonts";
+
+const Footer: React.FC = () => {
+  return (
+    <footer
+      className={`${OPEN_SANS_400.className} w-full text-sm leading-subtext text-subtext`}
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: use it in _app.tsx or

File: src/components/common/footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    >
+      <div className="container px-10 py-5 mx-auto">
+        <div className="flex relative justify-between items-center">
+          <div className="absolute w-full h-px bg-separator -top-5"></div>
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: remove absolute

File: src/components/common/footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import Image from "next/image";
+
+import { FOOTER_NAVIGATION, SOCIAL_LINKS } from "@/constants/links";
+
+import { OPEN_SANS_400 } from "@/styles/fonts";
+
+const Footer: React.FC = () => {
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: export

File: src/layouts/root-layout/RootLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+import { OPEN_SANS_400 } from "@/styles/fonts";
+import { ReactFCWithChildren } from "./types";
+
+const RootLayout: ReactFCWithChildren = ({ children }) => {
+  return (
+    <div
+      className={`${OPEN_SANS_400.className} flex flex-col justify-between h-screen`}
+    >
+      <Header />
```
Comment (Ilya-Valasiuk) [inline src/layouts/root-layout/RootLayout.tsx:7]: <Header />
<div className="container">
  <Page />

  <Footer class="bt-2 bt-grey" />
</Container>

File: src/components/common/header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import Image from "next/image";
+
+import { HEADER_NAVIGATION } from "@/constants/links";
+
+import { OPEN_SANS_600 } from "@/styles/fonts";
+import { filterClassNames } from "@/utils/filter-class-names";
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: classNames - https://www.npmjs.com/package/classnames

File: src/components/common/header/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            alt="Logo"
+            priority
+          />
+        </a>
+        <nav className="flex gap-x-8">
+          {HEADER_NAVIGATION.map(({ name, href, current }) => (
+            <a
+              key={name}
+              href={href}
+              className={filterClassNames(
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]:
```
className={classNames('text-white text-bold line-height-5', "sm:text-black", "md:text-green", "after:h-1")}
```

File: src/layouts/root-layout/RootLayout.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { Header } from "@/components/common/header/Header";
+import { Footer } from "@/components/common/footer/Footer";
+
+import { OPEN_SANS_400 } from "@/styles/fonts";
+import { ReactFCWithChildren } from "./types";
+
+const RootLayout: ReactFCWithChildren = ({ children }) => {
```
Comment (Ilya-Valasiuk) [inline src/layouts/root-layout/RootLayout.tsx]: const RootLayout: React.FC<React.PropsWithChildren>

File: src/components/common/footer/Footer.tsx (hunk 1 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Fixed

File: src/styles/fonts.ts (hunk 2 above)
Comment (a-reznikov) [reply src/styles/fonts.ts]: Fixed

File: src/components/common/footer/Footer.tsx (hunk 3 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Used

File: src/components/common/footer/Footer.tsx (hunk 4 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Fixed

File: src/components/common/footer/Footer.tsx (hunk 5 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Fixed

File: src/layouts/root-layout/RootLayout.tsx (hunk 6 above)
Comment (a-reznikov) [reply src/layouts/root-layout/RootLayout.tsx:7]: Fixed

File: src/components/common/header/Header.tsx (hunk 7 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Installed

File: src/components/common/header/Header.tsx (hunk 8 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Used

File: src/layouts/root-layout/RootLayout.tsx (hunk 9 above)
Comment (a-reznikov) [reply src/layouts/root-layout/RootLayout.tsx]: Fixed

**Review by Ilya-Valasiuk — APPROVED** [review]

**Pull request merged**